### ISSUE TITLE
[no-unused-vars] Recommended level should be "error"

### ISSUE THREAD

**Issue by Vanuan** [typescript-eslint/typescript-eslint#1320]
**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
import { something } from 'somepackage'

console.log('something is not used')
```

**Expected Result**

There should be a lint error

**Actual Result**

There's no lint error

**Additional Info**

Appears to be introduced in this commit:
https://github.com/typescript-eslint/typescript-eslint/commit/32753d1b07448ee4201edd17ff929bd00d55784f

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.10.0` |
| `@typescript-eslint/parser`        | `2.10.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.7.2` |
| `yarn`                              | `1.19.1` |

**Comment by bradzacher** [comment]
We do not have it set as an error on purpose.
Setting it as an error would make it build blocking for a large number of users.

However, there are a large number of false positives due to the implementation.
You are free to bump the severity in your own config if you are okay with the false positives blocking your CI.

Alternatively, try our new rule - `no-unused-vars-experimental`, which uses the same logic as the typescript compiler, so it doesn't have false positives.

**Comment by Vanuan** [comment]
ESLInt's level is error. I think it should be error in typescript too.

https://github.com/eslint/eslint/blob/dbbba83e3e81d53f7cfd4cb323a345f424ab0022/conf/eslint-recommended.js#L60-L61

**Comment by bradzacher** [comment]
It is, yes - eslint core only turns recommended rules on to error.

We will not be setting our recommendation to an error due to the false positives.

**Comment by Vanuan** [comment]
Got it. Thanks.

Are there any other recommended rules that are turned off?

**Comment by bradzacher** [comment]
We don't turn off any rules in the recommended set.

The exact recommended configs can be found here: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/configs

**Comment by Vanuan** [comment]
Trying to use it:
```
Error: Error while loading rule '@typescript-eslint/no-unused-vars-experimental': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
```

**Comment by Vanuan** [comment]
Documentation doesn't help:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#configuration

**Comment by Vanuan** [comment]
I think I got it.

```
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
  ],
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/no-unused-vars-experimental": "error",
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  }
}
```

Thank you.
